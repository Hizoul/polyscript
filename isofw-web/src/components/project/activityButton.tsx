import useActiveSetter from "isofw-shared/src/components/project/activeSetter"
import { observer } from "mobx-react"
import * as React from "react"
import WebButton from "../button"

@observer class ActiveButton extends React.Component<{id: string}, any> {
  public render() {
    const activeSetter = useActiveSetter(this.props.id)
    return (
      <WebButton
        text={activeSetter.isActive ? "Deactivate" : "Re-activate"}
        iconFa="step-backward"
        onClick={() => this.askActivation()}
        loading={activeSetter.loading}
      />
    )
  }
  private askActivation() {
    if (this.$f7 != null) {
      const activeSetter = useActiveSetter(this.props.id)
      if (activeSetter.isActive) {
        this.$f7.dialog.confirm(
          "This action will release all associated presets of this project! Are you sure you want to continue?",
          "Confirm", () => {
            activeSetter.toggleActive()
        })
      } else {
        activeSetter.toggleActive()
      }
    }
  }
}

export default ActiveButton
