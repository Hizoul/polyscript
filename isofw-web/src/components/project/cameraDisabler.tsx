import { Card, CardHeader, List, ListItem } from "framework7-react"
import useCameraDisabler from "isofw-shared/src/components/project/cameraDisabler"
import { IFieldProps } from "isofw-shared/src/util/xpfwform"
import { OperatorRelation, ProjectCameras } from "isofw-shared/src/xpfwDefs/project"
import { get } from "lodash"
import { observer } from "mobx-react-lite"
import * as React from "react"
import { CameraNameDisplayer, UserNameDisplayer } from "../displayName"

const WebCameraDisabler: React.FunctionComponent<IFieldProps & {autoSave?: boolean}> = observer((props) => {
  const disableHelper = useCameraDisabler(props.schema, props.mapTo, props.prefix, props.autoSave)
  return (
    <div className="flex row center">
      {disableHelper.operatorCameraMapping.map((item) => {
        const operatorId = get(item, String(OperatorRelation.title))
        return (
          <Card key={operatorId} style={{minWidth: "10rem"}}>
            <List>
              <ListItem
                onClick={disableHelper.operatorToggles[operatorId]}
              >
                <div slot="title">
                  <UserNameDisplayer id={operatorId} />
                </div>
                <div slot="inner">
                  <div className={`dot ${disableHelper.isActive[operatorId] === true ? "green" : "red"}`} />
                </div>
              </ListItem>
              {get(item, String(ProjectCameras.title), []).map((cameraId: any) => (
                <ListItem
                  key={cameraId}
                  onClick={disableHelper.cameraToggles[cameraId]}
                >
                  <div slot="title">
                    <CameraNameDisplayer id={cameraId} />
                  </div>
                  <div slot="inner">
                  <div className={`dot ${disableHelper.isActive[cameraId] === true ? "green" : "red"}`} />
                  </div>
                </ListItem>
              ))}
            </List>
          </Card>
        )
      })}
    </div>
  )
})

export default WebCameraDisabler
