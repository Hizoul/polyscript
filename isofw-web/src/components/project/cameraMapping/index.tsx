import { BlockTitle, Card, CardContent, CardHeader, Icon, List, ListItem, Popup, Row } from "framework7-react"
import SharedCameraMapping, { SharedCameraMappingProps } from "isofw-shared/src/components/project/cameraMapping"
import val from "isofw-shared/src/globals/val"
import { MailField } from "isofw-shared/src/util/xpfwuishared"
import { OperatorRelation, ProjectCameras, ProjectName } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import NameDisplayer from "isofw-web/src/components/displayName"
import { find, get } from "lodash"
import * as React from "react"
import "../style.sass"

const OperatorCamera: React.FunctionComponent<SharedCameraMappingProps & {operator: string}> = (props) => {
  return (
    <Card>
      <CardHeader>
        <NameDisplayer collection={val.service.user} id={props.operator} getNameFrom={MailField.mapTo} />
      </CardHeader>
      <List className="noMargin">
        <ul>
          {props.cameras.map((camera) => {
            const operatorCameras = find(props.value, [OperatorRelation.mapTo, props.operator])
            const selected = get(operatorCameras, ProjectCameras.mapTo, []).indexOf(camera) !== -1
            return <ListItem key={camera} onClick={props.changeMapping(props.operator, camera)}>
              <div slot="title">
                <NameDisplayer collection={val.service.camera} id={camera} getNameFrom={ProjectName.mapTo} />
              </div>
              <div slot="inner">
                <Icon fa={selected ? "times" : "plus"} color={selected ? "red" : "green"}/>
              </div>
            </ListItem>
          })}
        </ul>
      </List>
    </Card>
  )
}

const webCameraMapping: React.FunctionComponent<SharedCameraMappingProps> = (props) => {
  return (
    <div>
      <WebButton
        fill={true}
        big={true}
        onClick={props.togglePop}
        text="operator to camera mapping"
        iconFa="camera"
      />
      <Popup opened={props.showPopUp} onPopupClosed={props.togglePop}>
        <BlockTitle>select which operator handles which cameras</BlockTitle>
        {props.operators.map((operator) => <OperatorCamera {...props} key={operator} operator={operator} />)}
      </Popup>
    </div>
  )
}

const WrappedCameraMapping = SharedCameraMapping(webCameraMapping)
export default WrappedCameraMapping

