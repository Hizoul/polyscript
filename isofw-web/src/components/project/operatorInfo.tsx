import * as React from "react"
import { SharedFormEdit, IFormEditProps, DbStore, IFormShowProps } from "@xpfw/ui-shared";
import { ProjectForm, ProjectShot, ProjectProgram, ProjectName, ShotName, ShotType, ShotMovement, ShotMovementTowards, ShotDuration, ShotRemarksDirector, ShotRemarksOperator } from "isofw-shared/src/xpfwDefs/project";
import { FormStore, SharedField } from "@xpfw/form-shared";
import LoadingPage from "../loading";
import { get } from "lodash"
import { ComponentRegistry } from "@xpfw/form-shared"
import { FieldType } from "isofw-shared/src/util/xpfwvalidate";
import WebButton from "isofw-web/src/components/button";
import { Row, Col, Block, Link, Icon } from "framework7-react";
import "./style.sass"
import urls from "isofw-shared/src/globals/url";

const PresetItem: React.FunctionComponent<IFormShowProps> = (props) => {
  return (
    <tr>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>{get(props.item, ShotType.mapTo)}</td>
      <td>{get(props.item, ShotName.mapTo)}</td>
      <td>{get(props.item, ShotMovement.mapTo)}</td>
      <td>{get(props.item, ShotMovementTowards.mapTo)}</td>
      <td>{get(props.item, ShotDuration.mapTo)}</td>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>
        {get(props.item, ShotRemarksDirector.mapTo)}<br />
        {get(props.item, ShotRemarksOperator.mapTo)}
      </td>
    </tr>
  )
}

const OperatorInfo: React.FunctionComponent<IFormEditProps> = (props) => {
  console.log(" operator props ", props, get(props.original, ProjectProgram.mapTo, []))
  let i = 0
  const items = get(props.original, `result.${ProjectProgram.mapTo}`, []).map((item: any) => <PresetItem loading={false} item={item} key={++i} />)
  return (
    <div>
      <Row>
        <Col>
            <div className="subtitleBox withPadding">
              <Link href={urls.projectOverview} className="flex">
                <div className="flex1 column">
                  <div className="subtitle">active project</div>
                  {get(props, "original.result.name")}
                </div>
                <Icon fa="th-list" />
              </Link>
            </div>
        </Col>
        <Col>
          <div className="subtitleBox centerText">
            <div className="subtitle">Operator info</div>
            CA 5 6 9
          </div>
        </Col>
        <Col>
          <div className="subtitleBox withPadding">
            <div className="subtitle">Current shot</div>
            5
          </div>
        </Col>
      </Row>
      <Row className="marginTopBottom">
        <Col>
          <WebButton big fill text="shot ready" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="update preset" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="wide" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="stop zoom" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="tele" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="not ready" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="following position" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="script view" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
        <Col>
          <WebButton big fill text="preset view" iconFa="check" onClick={props.submitEdit} loading={props.loading} />
        </Col>
      </Row>
      <div className="data-table card" style={{marginLeft: "0pt", marginRight: "0pt"}}>
        <table>
          <thead>
            <tr>
              <th>shot</th>
              <th>CA</th>
              <th>Preset</th>
              <th>Type</th>
              <th>Object</th>
              <th>Movement</th>
              <th>Towards</th>
              <th>Duration</th>
              <th>picture</th>
              <th>remarks</th>
            </tr>
          </thead>
          <tbody>
            {items}
          </tbody>
        </table>
      </div>
      {props.loading ? <LoadingPage /> : null}
    </div>
  )
}

export default SharedFormEdit<{}>(OperatorInfo)