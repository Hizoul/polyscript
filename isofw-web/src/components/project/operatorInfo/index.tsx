import { Col, Icon, Link, Row } from "framework7-react"
import useOperatorInfo from "isofw-shared/src/components/project/operatorInfo"
import urls from "isofw-shared/src/globals/url"
import val from "isofw-shared/src/globals/val"
import { IEditHookProps } from "isofw-shared/src/util/xpfwdata"
import { ProjectName, ProjectShot, ShotCamera, ShotName } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import NameDisplayer from "isofw-web/src/components/displayName"
import { get } from "lodash"
import { observer } from "mobx-react-lite"
import * as React from "react"
import LoadingPage from "../../loading"
import "../style.sass"
import CameraPresets from "./cameraPresets"
import CurrentOperatorDisplay from "./currentOperatorDisplay"
import TablePresetItem from "./tablePresetItem"

const OperatorInfo: React.FunctionComponent<IEditHookProps> = observer((props) => {
  const previousPosition = React.useState(0)
  const operatorHelper = useOperatorInfo(props.id, props.mapTo, props.prefix)
  React.useEffect(() => {
    const newPosition = get(operatorHelper.original, String(ProjectShot.title))
    if (newPosition !== previousPosition[0]) {
      previousPosition[1](newPosition)
      const element = document.getElementById(`presetPositioner${newPosition}`)
      if (element != null) {
        element.scrollIntoView({
          behavior: "smooth", block: "center", inline: "center"
        })
      }
    }
  })
  let i = 0
  const content = operatorHelper.isPresetView ? (
    <div>
      {operatorHelper.presetByCamera.map((item: any) => {
            i++
            if (operatorHelper.currentCameras.length > 0 &&
              operatorHelper.currentCameras.indexOf(item.camera) === -1) {return null}
            return <CameraPresets item={item} key={item.camera} />
          })}
    </div>
  ) : (
    <div className="data-table card" style={{marginLeft: "0pt", marginRight: "0pt"}}>
      <table>
        <thead>
          <tr>
            <th>shot</th>
            <th>CA</th>
            <th>Preset</th>
            <th>!</th>
            <th>name</th>
            <th>Type</th>
            <th>Movement</th>
            <th>Towards</th>
            <th>Duration</th>
            <th>picture</th>
            <th>remarks</th>
          </tr>
        </thead>
        <tbody>
          {operatorHelper.filteredList.map((item: any) => {
            i++
            return <TablePresetItem item={item} key={i} />
          })}
        </tbody>
      </table>
    </div>
  )
  return (
    <div>
      <Row>
        <Col>
            <div className="subtitleBox withPadding">
              <Link href={urls.projectOverview} className="flex">
                <div className="flex1 column">
                  <div className="subtitle">active project</div>
                  {get(operatorHelper, "original.name")}
                </div>
                <Icon fa="th-list" />
              </Link>
            </div>
        </Col>
        <Col>
          <CurrentOperatorDisplay {...props} {...operatorHelper} />
        </Col>
        <Col>
          <div className="subtitleBox withPadding">
            <div className="subtitle">Current shot</div>
            {operatorHelper.currentPreset ? (
              <div>
                <NameDisplayer
                  collection={val.service.camera}
                  id={get(operatorHelper.currentPreset, String(ShotCamera.title))}
                  getNameFrom={String(ProjectName.title)}
                  placeholder=""
                /> &nbsp;
                {get(operatorHelper.currentPreset, String(ShotName.title), "")}
              </div>
            ) : null}
          </div>
        </Col>
      </Row>
      <Row className="marginTopBottom">
        <Col>
          <WebButton large={true} fill={true} text="shot ready" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="update preset" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="wide" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="stop zoom" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="tele" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="not ready" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton large={true} fill={true} text="following position" iconFa="check" onClick={operatorHelper.submitEdit} loading={operatorHelper.loading} />
        </Col>
        <Col>
          <WebButton
            large={true}
            fill={true}
            text={operatorHelper.isPresetView ? "script view" : "preset view"}
            iconFa={operatorHelper.isPresetView ? "list-ol" : "images"}
            onClick={operatorHelper.isPresetView ? operatorHelper.useScriptView : operatorHelper.usePresetView}
          />
        </Col>
      </Row>
      {content}
      {operatorHelper.loading ? <LoadingPage /> : null}
    </div>
  )
})
export default OperatorInfo
