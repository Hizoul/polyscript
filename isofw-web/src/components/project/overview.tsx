import { SharedFormList, IFormListProps, IFormShowProps } from "isofw-shared/src/util/xpfwuishared"
import * as React from "react"
import { get } from "lodash"
import { SharedField } from "@xpfw/form-shared"
import { ProjectName, ProjectForm } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "../button"
import urls from "isofw-shared/src/globals/url"

const ItemProject: React.FunctionComponent<IFormShowProps> = (props) => {
  return (
    <tr>
      <td>{get(props.item, ProjectName.mapTo)}</td>
      <td>
        <WebButton text="director" href={`${urls.directorPage}/${get(props.item, "_id")}`} />
        <WebButton text="program" href={`${urls.programPage}/${get(props.item, "_id")}`} />
        <WebButton text="operator" href={`${urls.operatorInfo}/${get(props.item, "_id")}`} />
        <WebButton text="edit" href={`${urls.edit}/${ProjectForm.collection}/${get(props.item, "_id")}`} />
      </td>
    </tr>
  )
}

const ProjectOverviewComponent: React.FunctionComponent<IFormListProps> = (props) => {
  const items = get(props, "list.result", []).map((item: any) => <ItemProject loading={false} item={item} />)
  return (
    <div className="data-table card">
      <table>
        <thead>
          <th>name</th>
          <th>actions</th>
        </thead>
        <tbody>
          {items}
        </tbody>
      </table>
      <WebButton text="Create" iconFa="plus" fill href={`${urls.create}/${ProjectForm.collection}`} style={{margin: "0.5rem"}} />
    </div>
  )
}

export default SharedFormList<{}>(ProjectOverviewComponent)