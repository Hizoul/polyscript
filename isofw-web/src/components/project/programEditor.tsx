import * as React from "react"
import { SharedFormEdit, IFormEditProps, DbStore } from "@xpfw/ui-shared";
import { ProjectForm, ProjectShot } from "isofw-shared/src/xpfwDefs/project";
import { FormStore } from "@xpfw/form-shared";
import sharedDirectorComponent, { DirectorComponentProps } from "isofw-shared/src/components/project/directorSheet";
import LoadingPage from "../loading";
import WebButton from "../button";
import { get } from "lodash"
import "./style.sass"
import { FaStepForward, FaStepBackward, FaSync } from "react-icons/fa";


const ProgramEditor: React.FunctionComponent<IFormEditProps> = (props) => {
  return (
    <div className="flex1 column">
      <div className="flex center">
        <div className="titleBox">
          {get(props, "original.result.name")}
        </div>
      </div>
      <div className="flex center">
        <div className="currentBox">
          <span className="shotNumber">{get(props, "original.result.shot")}</span>
          <br />
          <span>CA ?</span>
        </div>
      </div>
      {props.loading ? <LoadingPage /> : null}
    </div>
  )
}

export default SharedFormEdit<{}>(ProgramEditor)