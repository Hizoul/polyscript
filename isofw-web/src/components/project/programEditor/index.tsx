import { IEditHookProps, toJS, useEditWithProps } from "@xpfw/data"
import { prependPrefix, SharedField } from "@xpfw/form"
import { ProjectForm, ProjectProgram } from "isofw-shared/src/xpfwDefs/project"
import WebButton from "isofw-web/src/components/button"
import { get } from "lodash"
import { observer } from "mobx-react-lite"
import * as React from "react"

const programTheme = "program"

const ProgramEditor: React.FunctionComponent<IEditHookProps> = observer((props) => {
  const editHelper = useEditWithProps(props)
  return (
    <div className="flex1 column">
      <div className="flex center" style={{marginTop: "0.4rem", marginBottom: "1.5rem"}}>
        <div className="titleBox">
          {get(editHelper, "original.name")}
        </div>
      </div>
      <SharedField schema={ProjectProgram} prefix={prependPrefix(ProjectForm.title, props.prefix)} theme={programTheme} />
    </div>
  )
})

export default ProgramEditor
