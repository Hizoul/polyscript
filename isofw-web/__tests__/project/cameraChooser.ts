import cameraChooserTest from "isofw-shared/src/tests/project/cameraChooser"
import WebCameraChoice from "isofw-web/src/components/project/programEditor/cameraChooser";
import { ComponentRegistry } from "@xpfw/form-shared";
import { FieldType } from "@xpfw/validate";
import { ShotCamera } from "isofw-shared/src/xpfwDefs/project"

ComponentRegistry.registerComponent(FieldType.RelationshipSingle, WebCameraChoice, ShotCamera.theme)
cameraChooserTest()
