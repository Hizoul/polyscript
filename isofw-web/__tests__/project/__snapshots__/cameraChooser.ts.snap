// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` camera mapping test  works as planned:  after Closing pop up  1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        Camera#2
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": false
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#1
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#2
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#3
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` camera mapping test  works as planned:  after Fetching cameras  1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        Camera#2
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": false
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#1
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#2
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#3
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` camera mapping test  works as planned:  after Opening pop up  1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        Camera#2
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": true
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#1
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#2
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span>
            Camera#3
          </span>
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` camera mapping test  works as planned:  after Setting Camera  1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        not yet selected
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": false
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` camera mapping test  works as planned:  after preparing cameras  1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        not yet selected
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": false
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="inner"
        >
          <div
            className="Icon"
          >
            {
  "fa": "chevron"
}
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` camera mapping test  works as planned: Before anything 1`] = `
<div>
  <div
    className="ListItem"
  >
    {}
    <div
      slot="title"
    >
      Camera: 
      <span>
        not yet selected
      </span>
    </div>
    <div
      slot="inner"
    >
      <div
        className="Icon"
      >
        {
  "fa": "camera"
}
      </div>
    </div>
  </div>
  <div
    className="Popup"
  >
    {
  "opened": false
}
    <div
      className="BlockTitle"
    >
      {}
      Choose a camera 
      camera
    </div>
    <div
      className="List"
    >
      {}
    </div>
  </div>
</div>
`;
